index.js

```diff
@@ -482,6 +487,31 @@ async function findSchemaPaths(schemaBasePath, options = {}) {
     .map(p => path.join(p.dir, p.base));
 }
 
+
+/**
+ * Compare function for schema info, used for sorting based
+ * on 'common' schema, title, semver, and 'current'.
+ * There's no good way to know of schema dependency order without
+ * building a graph, but we can at least guess with some good heuristics.
+ *
+ * @param {Object} infoA
+ * @param {Objectt} infoB
+ * @return {int}
+ */
+function schemaInfoCompare(infoA, infoB) {
+    // titles with 'common' in them should sort earlier.
+    // (If common is in the title, assume it is likely a dependency schema.)
+    const infoACommon = infoA.title.includes('common');
+    const infoBCommon = infoB.title.includes('common');
+    return infoACommon === infoBCommon ? 0 : (infoACommon ? -1 : 1) ||
```

Comment:
Perhaps if both have 'common' we can have the one with a shorter path (shorter in terms of number of subdirectories) go first?

Reply:
I can add hierarchy depth too.  Question is what is the better valid hierarchy:  The actual filepath hierarchy, or the title hierarchy?  I use filepath hierarchy above in the materializeModifiedSchemas because I don't parse title at that point.  Perhaps I should?  Title path seems better, as it is more canonical (that's why I use version from $id instead of filename).

Reply:
Either way it would be best to be consistent at when and how I sort before I generate.  Hm.  I will use path hierarchy just to be consistent.

Reply:
Done.